SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -171,47 +191,50 @@ private static int CreateDeployment(string[] args, bool useChinaPlatform)
                 // to be able to start the deployment. The sim players don't care about the actual snapshot.
-                var createSimDeploymentOp = CreateSimPlayerDeploymentAsync(deploymentServiceClient,
-                    projectName, assemblyName, runtimeVersion, mainDeploymentName, simDeploymentName,
+                var simDeploymentCreationOps = CreateSimPlayerDeploymentsAsync(deploymentServiceClient,
+                    projectName, assemblyName, runtimeVersion, mainDeploymentName, simDeploymentBaseName,
                     simDeploymentJson, mainDeploymentSnapshotPath, simDeploymentRegion, simDeploymentCluster,
-                    simNumPlayers, useChinaPlatform);
+                    numSimPlayers, numSimDeployments, useChinaPlatform);
 
-                // Wait for both deployments to be created.
-                Console.WriteLine("Waiting for deployments to be ready...");
+                if (simDeploymentCreationOps == null || simDeploymentCreationOps.Count == 0)
+                {
+                    Console.WriteLine("Failed to start any simulated player deployments.");
+                    return 1;
+                }
+
+                // Wait for the main deployment to be ready
+                Console.WriteLine("Waiting for the main deployment to be ready...");
                 var mainDeploymentResult = createMainDeploymentOp.PollUntilCompleted().GetResultOrNull();
                 if (mainDeploymentResult == null)
                 {
                     Console.WriteLine("Failed to create the main deployment");
                     return 1;
                 }
-
                 Console.WriteLine("Successfully created the main deployment");
-                var simPlayerDeployment = createSimDeploymentOp.PollUntilCompleted().GetResultOrNull();
-                if (simPlayerDeployment == null)
+
+                // Waiting for the simulated player deployment(s) to be ready
+                var numSuccessfullyStartedSimDeployments = 0;
+                for (var simDeploymentIndex = 0; simDeploymentIndex < simDeploymentCreationOps.Count; simDeploymentIndex++)
                 {
-                    Console.WriteLine("Failed to create the simulated player deployment");
-                    return 1;
-                }
+                    Console.WriteLine($"Waiting for the simulated player deployment to be ready... " +
+                        $"(deployment {simDeploymentIndex + 1}/{simDeploymentCreationOps.Count})");
 
-                Console.WriteLine("Successfully created the simulated player deployment");
+                    var simPlayerDeployment = simDeploymentCreationOps[simDeploymentIndex].PollUntilCompleted().GetResultOrNull();
+                    if (simPlayerDeployment == null)
+                    {
+                        Console.WriteLine($"Failed to create the simulated player deployment");
```

Comment:
Worth noting the index here of which one failed.

Reply:
The index for the "current" deployment is mentioned in https://github.com/spatialos/UnrealGDK/pull/2398/files#diff-893453b6043e557819612fff6168a107R221

Do you think it could use another mention here?

Reply:
yeah,  I think it's always good to have another mention of it in the line where it actually fails.